File: TenMinsJava/day97.java
// press 'u' to update a row, ask for a rowId, then ask for the field values (day 82)
package TenMinsJava;

import java.io.File;
import java.io.FileWriter;
import java.util.ArrayList;
import java.util.Scanner;

public class day97 {
    public static void updateCsvCell(int rowId, int columnNumber, String value) {
        try {
            File file = new File("D:/coding30minsAday/textFiles/JTrialCsv.csv");
            Scanner fScan = new Scanner(file);
            ArrayList<String> lines = new ArrayList<>();

            // Step 1: Read all lines
            while (fScan.hasNextLine()) {
                lines.add(fScan.nextLine());
            }
            fScan.close();

            // modifies the value
            // try lines.set()
            if (lines.size() >= rowId) {
                String[] cells = lines.get(rowId - 1).split(",");
                if (columnNumber <= cells.length) {
                    cells[columnNumber] = String.valueOf(value);
                    StringBuilder b = new StringBuilder();
                    for (int i = 0; i < cells.length; i++) {
                        b.append(cells[i].trim());
                        if (i < cells.length - 1) b.append(",");
                    }
                    lines.set(rowId - 1, b.toString());
                }
            }

            // rewrite the csv
            FileWriter w = new FileWriter(file);
            for (String string : lines) {
                w.write(string + "\n");
            }
            w.close();
        } catch (Exception e) {
            System.err.println("Error");
        }
    }
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        System.out.println("Press u to update row");
        String choice = scan.nextLine();
        if (choice.trim().toLowerCase().equals("u")) {
            System.out.println("Enter row id");
            String rowId = scan.nextLine().trim();
            System.out.println("Enter column id");
            String columnId = scan.nextLine().trim();
            System.out.println("Enter new value");
            String value = scan.nextLine().trim();
            int newRowId = Integer.parseInt(rowId);
            int newColumnId = Integer.parseInt(columnId);
            updateCsvCell(newRowId, newColumnId, value);
        }
        scan.close();
    }
}

File: TenMinsPy/day97.py
# press 'u' to update a row, ask for a rowId, then ask for the field values (day 82)

import csv

def update_csv_cell(row_id, column_number, value, file_path="D:/coding30minsAday/textFiles/JTrialCsv.csv"):
    try:
        # Step 1: Read all lines
        with open(file_path, "r") as f:
            lines = f.readlines()

        # Step 2: Modify the value
        if len(lines) >= row_id:
            cells = lines[row_id - 1].strip().split(",")
            if column_number <= len(cells):
                cells[column_number - 1] = str(value)  # Python is 0-indexed
                lines[row_id - 1] = ",".join(c.strip() for c in cells) + "\n"

        # Step 3: Rewrite the CSV
        with open(file_path, "w") as f:
            f.writelines(lines)

    except Exception as e:
        print("Error:", e)

def ask():
    choice = input("enter u to update row\n").strip()
    if choice.lower() == "u":
        row_id = int(input("enter row id number"))
        column_number = int(input("enter column id number"))
        value = (input("enter new value"))

ask()